client/src/utils/fetch-api.js

```diff
@@ -0,0 +1,25 @@
+import Axios from 'axios';
+import queryParams from 'qs';
+import { apiUrl } from '../config/environment';
+
+Axios.defaults.baseURL = apiUrl;
+Axios.defaults.paramsSerializer = params => queryParams.stringify(params);
+
+const fetchAPI = options => new Promise((resolve, reject) => {
+  Axios(options)
+    .then(resolve)
+    .catch(error => reject(error.response.data));
+});
+
+const buildRequest = method => url => options => fetchAPI({
+  method,
+  url,
+  ...options,
```

Comment:
are options same for all type of requests?
Because when I implemented this thing using fetch api then get options and post options were different

Reply:
You can call any type of request using same function. 

```js
Axios({
url, method, data, params, etc
})
```


The other format is to use Axios.post and Axios.get method which I'm not using because I wanted to create a more general request methods. 

**are options same for all type of requests?**  

From what I've read, POST, PUT and PATCH method require `data` whereas others methods do not. Other than that, I don't think there's any difference.